src/algebra/algebra/tower.lean

```diff
@@ -184,6 +184,11 @@ lemma restrict_scalars_apply (f : A →ₐ[S] B) (x : A) : f.restrict_scalars R
 
 @[simp] lemma coe_restrict_scalars' (f : A →ₐ[S] B) : (restrict_scalars R f : A → B) = f := rfl
 
+lemma restrict_scalars_injective :
+  function.injective (restrict_scalars R : ((A →ₐ[S] B) → (A →ₐ[R] B))) :=
+λ f g h, alg_hom.ext (λ x,
+show f.restrict_scalars R x = g.restrict_scalars R x, from alg_hom.congr_fun h x)
```

Comment:
```suggestion
λ f g h, alg_hom.ext (alg_hom.congr_fun h : _)
```

Reply:
Oh yeah, the `(_: _)` trick for confusing the elaborator. Nice idea!